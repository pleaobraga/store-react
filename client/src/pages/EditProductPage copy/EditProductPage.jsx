/* eslint-disable react/prop-types */
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getProduct } from '../../reducers/ProductReducer'
import { useParams } from 'react-router'

import Loading from '../../components/Loading'
import ErrorPage from '../ErrorPage'
import EditProduct from '../../components/EditProduct'

import './EditProductPage.scss'

const EditProductPage = () => {
  const dispatch = useDispatch()
  const pageContent = useSelector(state => state.product)
  const { id } = useParams()

  React.useEffect(() => {
    dispatch(getProduct(id))
  }, [])

  const contentPage = ({ product, title }) => {
    return (
      <div className="page page-edit-product">
        <h1 className="page-edit-product__title">{title}</h1>
        <EditProduct currentProduct={product} isEditing={true} />
      </div>
    )
  }

  const { errorContent, product } = pageContent

  return errorContent ? (
    <ErrorPage />
  ) : product ? (
    contentPage({ product, title: 'Edit Product' })
  ) : (
    <Loading />
  )
}

export default EditProductPage
