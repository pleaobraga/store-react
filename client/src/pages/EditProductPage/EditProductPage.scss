.page {
  &-edit-product {
    align-items: flex-start;
    flex-direction: column;
    justify-content: flex-start;

    &__title {
      font-size: 7rem;
      margin-bottom: 7rem;
    }
  }
}
