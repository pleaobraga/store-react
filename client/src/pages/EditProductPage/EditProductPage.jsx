/* eslint-disable react/prop-types */
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getProduct } from '../../reducers/ProductReducer'
import { useParams } from 'react-router'

import Loading from '../../components/Loading'
import ErrorPage from '../ErrorPage'
import EditProduct from '../../components/EditProduct'

import './EditProductPage.scss'

const EditProductPage = () => {
  const dispatch = useDispatch()
  const pageContent = useSelector(state => state.product)
  const { id } = useParams()

  React.useEffect(() => {
    if (id !== 'new') dispatch(getProduct(id))
  }, [])

  const contentPage = ({ product, title }) => {
    return (
      <div className="page page-edit-product">
        <h1 className="page-edit-product__title">{title}</h1>
        <EditProduct {...product} />
      </div>
    )
  }

  if (id === 'new') {
    return contentPage({ product: {}, title: 'Create new Product' })
  }

  const { errorContent, product } = pageContent

  return errorContent ? (
    <ErrorPage />
  ) : product ? (
    contentPage({ product, title: 'Edit Product' })
  ) : (
    <Loading />
  )
}

export default EditProductPage
