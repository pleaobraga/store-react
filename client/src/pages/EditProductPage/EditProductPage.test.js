import React from 'react'
import { mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'

import {
  mockStore,
  initialStateReducer,
  product
} from '../../reducers/__mocks__/reduxMock'
import EditProductPage from './EditProductPage'

const setup = ({ state }) => {
  const newState = {
    product: {
      ...initialStateReducer,
      ...state
    }
  }

  const store = mockStore(newState)

  const contentPage = mount(
    <Provider store={store}>
      <Router>
        <EditProductPage />
      </Router>
    </Provider>
  )

  return contentPage
}

describe('EditProductPage', () => {
  it('should render content page', () => {
    const editProductPage = setup({ product })
    expect(editProductPage).toMatchSnapshot()
  })

  it('should render error page', () => {
    const editProductPage = setup({ errorContent: true })
    expect(editProductPage).toMatchSnapshot()
  })

  it('should render loader ', () => {
    const editProductPage = setup({ loadingContent: true })
    expect(editProductPage).toMatchSnapshot()
  })
})
