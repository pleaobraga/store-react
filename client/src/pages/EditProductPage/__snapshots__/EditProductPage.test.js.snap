// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProductPage should render content page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <EditProductPage>
        <Loading>
          <div
            className="loading undefined"
          >
            <LoaderIcon
              color="#298a95"
            >
              <svg
                className="lds-spinner"
                fill="#298a95"
                height="200px"
                preserveAspectRatio="xMidYMid"
                style={
                  Object {
                    "background": "none",
                  }
                }
                viewBox="0 0 100 100"
                width="200px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <g
                  transform="rotate(0 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.9166666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(30 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.8333333333333334s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(60 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.75s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(90 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.6666666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(120 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.5833333333333334s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(150 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.5s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(180 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.4166666666666667s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(210 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.3333333333333333s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(240 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.25s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(270 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.16666666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(300 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.08333333333333333s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(330 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="0s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
              </svg>
            </LoaderIcon>
          </div>
        </Loading>
      </EditProductPage>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`EditProductPage should render error page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <EditProductPage>
        <Loading>
          <div
            className="loading undefined"
          >
            <LoaderIcon
              color="#298a95"
            >
              <svg
                className="lds-spinner"
                fill="#298a95"
                height="200px"
                preserveAspectRatio="xMidYMid"
                style={
                  Object {
                    "background": "none",
                  }
                }
                viewBox="0 0 100 100"
                width="200px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <g
                  transform="rotate(0 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.9166666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(30 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.8333333333333334s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(60 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.75s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(90 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.6666666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(120 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.5833333333333334s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(150 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.5s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(180 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.4166666666666667s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(210 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.3333333333333333s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(240 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.25s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(270 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.16666666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(300 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.08333333333333333s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(330 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="0s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
              </svg>
            </LoaderIcon>
          </div>
        </Loading>
      </EditProductPage>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`EditProductPage should render loader  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <EditProductPage>
        <Loading>
          <div
            className="loading undefined"
          >
            <LoaderIcon
              color="#298a95"
            >
              <svg
                className="lds-spinner"
                fill="#298a95"
                height="200px"
                preserveAspectRatio="xMidYMid"
                style={
                  Object {
                    "background": "none",
                  }
                }
                viewBox="0 0 100 100"
                width="200px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <g
                  transform="rotate(0 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.9166666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(30 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.8333333333333334s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(60 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.75s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(90 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.6666666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(120 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.5833333333333334s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(150 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.5s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(180 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.4166666666666667s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(210 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.3333333333333333s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(240 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.25s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(270 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.16666666666666666s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(300 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="-0.08333333333333333s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
                <g
                  transform="rotate(330 50 50)"
                >
                  <rect
                    fill="#298a95"
                    height="12"
                    rx="9.4"
                    ry="4.8"
                    width="6"
                    x="47"
                    y="24"
                  >
                    <animate
                      attributeName="opacity"
                      begin="0s"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="1;0"
                    />
                  </rect>
                </g>
              </svg>
            </LoaderIcon>
          </div>
        </Loading>
      </EditProductPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
